src/heretic/model.py

```diff
@@ -58,9 +61,25 @@ def __init__(self, settings: Settings):
             print(f"* Trying dtype [bold]{dtype}[/]... ", end="")
 
             try:
+                quantization_config = None
+                if settings.load_in_4bit:
+                    # BitsAndBytesConfig expects a torch.dtype, not a string.
+                    if dtype == "auto":
+                        compute_dtype = torch.bfloat16
```

Comment:
But we still pass the original string to `AutoModelForCausalLM.from_pretrained` below. What if `dtype == "auto"` and the `dtype` in the model's configuration is something other than BF16? Won't that cause problems since the quantization config doesn't match the `dtype`?